### PR TITLE
Fixed wrong relational simplifications and added tests

### LINKED ISSUE #19407: Simplification gives wrong results
I am trying to simplify a simple equation:

```
import sympy

q, r = sympy.symbols("q r")
equation = (((-q + r) - (q - r)) <= 0).simplify()
print(equation) # q >= r

equation = ((sympy.sqrt(2) * (-q + r) - sympy.sqrt(2) * (q - r)) <= 0).simplify()
print(equation) # q <= r
```

As you can see from the comments, the results differ; but they should not, if I see that correctly. I am using Sympy version 1.5.1 and this happens on either Python 3.6.6 and 3.7.7. See also [this SO question](https://stackoverflow.com/questions/61965660/does-sympy-give-errorneous-simplifications). As was pointed out in the comments, this also happens for other coefficients other thant `sympy.sqrt(2)`.

### PR DESCRIPTION
**Pull request by KaustubhDamania**
In earlier version, the following simplification gave wrong result:
```python
>> import sympy
>> q, r = sympy.symbols("q r")
>> root2 = sympy.sqrt(2)
>> ((root2 * (-q + r) - root2 * (q - r)) <= 0).simplify()
q <= r
```

The correct output should have been ```q >=r```.
This issue was raised at #19407. I fixed this issue by ensuring
```sympy.polys.gcd``` always returned a non-negative number, which was
called inside ```eval_simplify()``` in ```sympy/core/relational.py```. I have
also added relevant tests.

#### References to other Issues or PRs
Fixes #19407

#### Brief description of what is fixed or changed
Modified ```gcd_list()``` and ```gcd()``` in ```sympy/polys/polytools.py``` to return non-negative constant.
Added tests in ```sympy/core/tests/test_relational.py``` and ```sympy/polys/tests/test_polytools.py```.

#### Release Notes

* polys
  * Fixed GCD to always return a non-negative constant.

### CODE REVIEW

File: sympy/polys/tests/test_polytools.py [hunk 1]
```diff
@@ -1900,6 +1900,8 @@ def test_gcd_list():
     gcd = gcd_list([], x, polys=True)
     assert gcd.is_Poly and gcd.is_zero
 
+    assert gcd_list([sqrt(2), -sqrt(2)]) == sqrt(2)
```
Comment (smichr) [inline sympy/polys/tests/test_polytools.py:1903]: This one passes for me in master but the reverse fails. Perhaps just include both:
```
s = sqrt(2)
assert gcd_list([a, -a]) == gcd_list([-a, a]) == s
```

File: sympy/polys/tests/test_polytools.py [hunk 2]
```diff
@@ -1993,6 +1995,12 @@ def test_gcd():
     assert gcd(f, g, modulus=11, symmetric=False) == h
     assert lcm(f, g, modulus=11, symmetric=False) == l
 
+    a, b = sqrt(2), -sqrt(2)
```
Comment (smichr) [inline sympy/polys/tests/test_polytools.py:1998]: It might be a good idea to confirm that the reverse order passes, too. (And it does with this PR.)

**Comment by smichr** [comment]
Just a few suggestions on tests, otherwise this looks good.

File: sympy/polys/tests/test_polytools.py (hunk 1 above)
Comment (KaustubhDamania) [reply sympy/polys/tests/test_polytools.py:1903]: Yeah, the reverse is also passing for me, I'll add those changes you suggested.

**Comment by smichr** [comment]
Thanks. It's in.

**Pull request merged**